prov/sm2/src/sm2_coordination.c

```diff
@@ -0,0 +1,675 @@
+		peer_pid = entries[jentry].pid;
+		if (peer_pid == 0)
+			goto found;
+		else if (peer_pid < 0) {
+			// A third peer might have entered this address into their AV,
+			// and there is no current way to check this... need to keep this
+			// entry in the file until we clean up
+			continue;
+		}
+		else {
+			if (!pid_lives(peer_pid)) {
+				sm2_coordinator_extend_for_entry(map, jentry);
+				entries = sm2_mmap_entries(map);
+				header = (void *)map->base;
+				struct sm2_region *peer_region =
+				    sm2_mmap_ep_region(map, jentry);
+
+				if (entries[jentry].startup_ready &&
+				    smr_freestack_isfull(sm2_free_stack(peer_region))) {
+					/* we found a slot with a dead PID and the
+					 * freestack is full */
+					entries[jentry].pid = 0;
+					goto found;
+				};
+			}
+		}
+	}
+
+	fprintf(stderr,
+		"No available entries were found in the coordination file, all %d were "
+		"used\n",
+		header->ep_enumerations_max);
+	return -FI_EAVAIL;
+
+found:
+
+	if (self) {
+		entries[jentry].startup_ready = 0;
+		atomic_wmb();
+		entries[jentry].pid = pid;
+	}
+
+	if (!self && entries[jentry].pid == 0) {
+		entries[jentry].startup_ready = 0;
+		atomic_wmb();
+		entries[jentry].pid = -pid;
+	}
+
+	FI_WARN(&sm2_prov, FI_LOG_AV, "Allocated sm2 region for %s at AV[%d]\n", name,
+		jentry);
+	strncpy(entries[jentry].ep_name, name, SM2_NAME_MAX);
+
+	*av_key = jentry;
+
+	/* With the entry allocated, we now need to ensure it's mapped. */
+	if (!sm2_mapping_long_enough_check(map, jentry)) {
+		sm2_coordinator_extend_for_entry(map, jentry);
+	}
+	return 0;
+}
+
+/**
+ * @brief look-up the name.
+ *
+ * @param[in] name
+ * @param[in] map
+ * @return the index of the name, or -1
+ */
+int sm2_coordinator_lookup_entry(const char *name, struct sm2_mmap *map)
+{
+
+	struct sm2_coord_file_header *header = (void *)map->base;
+	struct sm2_ep_allocation_entry *entries;
+	int jentry;
+
+	entries = sm2_mmap_entries(map);
+
+	for (jentry = 0; jentry < header->ep_enumerations_max; jentry++) {
+		if (0 == strncmp(name, entries[jentry].ep_name, SM2_NAME_MAX)) {
+			FI_WARN(&sm2_prov, FI_LOG_AV, "Found existing %s in slot %d\n",
+				name, jentry);
+			return jentry;
+		}
+		if (entries[jentry].ep_name[0] != '\0') {
+			FI_WARN(&sm2_prov, FI_LOG_EP_CTRL,
+				"Searching for %s. Not yet found in spot AV[%d]=%s, PID= "
+				"%d\n",
+				name, jentry, entries[jentry].ep_name,
+				entries[jentry].pid);
+		}
+		else if (!strcmp(entries[jentry].ep_name, ZOMBIE_ALLOCATION_NAME)) {
+			FI_WARN(&sm2_prov, FI_LOG_EP_CTRL,
+				"Found zombie in spot AV[%d], PID= %d\n", jentry,
+				entries[jentry].pid);
+		}
+	}
+	return -1;
+}
+
+/**
+ * @brief clear the pid for this entry.  must already hold lock.
+ *
+ * @param[in] name
+ * @param[inout] map
+ * @param[out] av_key
+ */
+ssize_t sm2_coordinator_free_entry(struct sm2_mmap *map, int av_key)
+{
+	struct sm2_ep_allocation_entry *entries;
+	int pid = getpid();
+	(void)pid;
```

Comment:
does this do anything?

Reply:
Done. It was stopping a compile warning when compiling without asserts, unused variable.  Got ride of variable.